linker/descriptors.go

```diff
@@ -203,14 +202,6 @@ func asSourceLocations(srcInfoProtos []*descriptorpb.SourceCodeInfo_Location) []
 	return locs
 }
 
-func pathStr(p protoreflect.SourcePath) string {
-	var buf bytes.Buffer
-	for _, v := range p {
-		_, _ = fmt.Fprintf(&buf, "%x:", v)
-	}
-	return buf.String()
-}
-
```

Comment:
This wasn't really needed. We already had a `pathKey` function, to turn the path (a slice, which can't be used as a map key) into a value that can be used as a key.